config/src/main/java/com/networknt/config/ConfigInjection.java

```diff
@@ -160,4 +169,29 @@ public void setKey(String key) {
             this.key = key;
         }
     }
+
+    // Method used to cast string into int, double or boolean
+    private static Object typeCast(String str) {
+        if (str == null || str.equals("")) {
+            return null;
+        }
+        if (isBoolean(str)) {
+            return str.equals("true") || str.equals("True") || str.equals("TRUE") ? true : false;
+        }
```

Comment:
Can be simplified to just `return str.equals("true") || str.equals("True") || str.equals("TRUE");` right?

Reply:
or
return str.toLowerCase().equals("true");

Reply:
Yes, I will simplify it. @miklish The reason why I didn't use toLowerCase() is to uniform with yaml rules, ie "tRue", "trUe", etc. should not be recognized as boolean. Do you think it makes sense?

Reply:
It simply depends on how strict you want your parsing to be in this instance.

Reply:
@jiachen1120 Agreed, we shouldn't use toLowerCase.
Actually, if you want to be even closer to the [yaml boolean spec](https://yaml.org/type/bool.html), we would use these:
```
 y|Y|yes|Yes|YES|n|N|no|No|NO
|true|True|TRUE|false|False|FALSE
|on|On|ON|off|Off|OFF
```

Reply:
Sure, I will improve it to be more strict.

Reply:
And I will modify the light-doc as well.